rest-service/manager_rest/deployment_update/manager.py

```diff
@@ -573,6 +608,60 @@ def _execute_workflow(self,
         )
         return new_execution
 
+    def _extract_plugins_changes(self, dep_update):
+        """Extracts plugins that need to be installed and uninstalled.
+
+        :param dep_update: a DeploymentUpdate object.
+        :return: plugins that need installation and uninstallation (a tuple).
+        """
+
+        def is_centrally_deployed(plugin):
+            return plugin[constants.PLUGIN_EXECUTOR_KEY] \
+                   == constants.CENTRAL_DEPLOYMENT_AGENT
+
+        def append_plugins_to_list(source_dict, key_list, target_list):
+            for k in key_list:
+                target_list.append(source_dict[k])
+
+        deployment = self.sm.get(models.Deployment, dep_update.deployment_id)
+        old_plan = deployment.blueprint.plan
+        new_plan = dep_update.deployment_plan
+        # Plugins to install in the old plan
+        plugins_to_install_old = extract_and_merge_plugins_to_install(
+            old_plan[constants.DEPLOYMENT_PLUGINS_TO_INSTALL],
+            old_plan[constants.WORKFLOW_PLUGINS_TO_INSTALL],
+            filter_func=is_centrally_deployed)
+        # Plugins to install in the new plan
+        plugins_to_install_new = extract_and_merge_plugins_to_uninstall(
+            new_plan[constants.DEPLOYMENT_PLUGINS_TO_INSTALL],
+            new_plan[constants.WORKFLOW_PLUGINS_TO_INSTALL],
+            filter_func=is_centrally_deployed)
+        # Convert to plugin_name->plugin dict
+        new_plugins = {p[constants.PLUGIN_NAME_KEY]: p
+                       for p in plugins_to_install_new}
+        old_plugins = {p[constants.PLUGIN_NAME_KEY]: p
+                       for p in plugins_to_install_old}
+        new_p_names = set(new_plugins)
+        old_p_names = set(old_plugins)
```

Comment:
may not merge this handling from 640-645 to the relevant extract_and_merge_plugins_to_uninstall

Reply:
I don't think this belongs to the `extract_and_merge` function. This is a comparison logic between two lists of plugins, and `extract_and_merge` only deals with merging workflow + deployment plugins. Your thoughts?